demoapp/demo_app_functions.py

```diff
@@ -450,4 +466,479 @@ def update_csv(bill_num: str, title: str, summarized_bill: str, category: str, t
+                
+                Also use committee information {getattr(bill_details, 'committee_info')} if available.\n
+
+                INSTRUCTIONS: 
+                1. Choose minimum of 3 tags and no more than 5.
+                2. Do not provide explanations for the tag choices.
+                3. Do not output tags not listed above.
+                4. Do not modify or paraphrase the rag names, choose directly from the list provided.
+
+                Response Format:
+                
+                Tags:TAGS
+                """
+    else: 
+
+        query = """ Your Job here is to identify the tags from various categories that can be associated to the following MA Legislative bill. 
+
+                Note that the bill refers to specific existing chapters and sections of the Mass General Laws. Use the information from those chapters and sections in your context for tagging.
+                
+                List of tags: 
+                - {category_tags}
+
+                The bill title is: \"{title}"\
+
+                The bill text is: \"{context}"\
+                
+                The relevant committee information: \"{committee_info}"\
+
+                The relevant section names are: \"{names}"\
+
+                The relevant section text is: \"{mgl_sections}"
+
+                INSTRUCTIONS: 
+                1. Choose minimum of 3 tags and no more than 5.
+                2. Do not provide explanations for the tag choices.
+                3. Do not output tags not listed above.
+                4. Do not modify or paraphrase the rag names, choose directly from the list provided.
+
+                Response Format:
+                
+                Tags:TAGS
+
+                """
+        bill_details.invoke_dict = prepare_invoke_dict(bill_details)
+        bill_details.invoke_dict['category_tags'] = category_tags
+
+    return query
+
+def call_llm(bill_details: BillDetails, query: str, llm_call_type: str = 'small', return_metrics: bool = False) -> LLMResults: 
+
+    """
+        
+    This is a generic function that calls the LLM with given query
+
+    Args: 
+        bill_details (BillDetails): Object containing information about the bill - bill_text, bill_title, mgl_ref, commottee_info, mgl_names
+        query (str): Query string containing details on what model has to do. 
+        llm_call_type (str): This argument can take 2 values ("small": No use of RAG, "large": Use RAG) 
+        return_metrics (bool): This argument specifies weather to calculate metrics like rouge socores, cosine similarity and factual similarity or not. 
+
+    Returns: 
+        LLMResults: Object containing query, response (Raw unformatted response from model) and metrics (If requested)
+
+    """
+
+    API_KEY = getattr(bill_details, "OPENAI_API_KEY")
```

Comment:
I suspect the OPEN_API_KEY env variable can be assumed to be set already in tnhe container of the lambda function